frontend/src/pages/NewAgreement.tsx

```diff
@@ -111,22 +136,97 @@ const NewAgreement: FC = () => {
   /**
    * Rationale
    */
-  const [rationale, setRationale] = useState(''); //@todo default to value in extant record if one exists
-  const [isRationaleEdited, setIsRationaleEdited] = useState(false); //@todo set true when name is loaded from extant record in Hasura
+  const [rationale, setRationale] = useState('');
+  const [isRationaleEdited, setIsRationaleEdited] = useState(false);
 
   /**
-   * "Continue" and insert new agreement
+   * Agreement insert
    */
   const [
     addAgreement,
     { data: addAgreementData, loading: addAgreementLoading, error: addAgreementError },
-  ] = useMutation(ADD_AGREEMENT, { refetchQueries: ['agreements'] });
+  ] = useMutation(addAgreementMutation, {
+    refetchQueries: ['agreements'],
+    onCompleted: (data) => {
+      // When new agreement is saved and new record id comes back from hasura, update state and sessionStorage
+      if (data?.insert_core_agreements_one.id && agreementId === null) {
+        const newAgreementId = data.insert_core_agreements_one.id;
+        sessionStorage.setItem(sessionStorageKey, newAgreementId as unknown as string);
+        setAgreementId(newAgreementId);
+        setSavedData(data.insert_core_agreements_one);
+      }
+    },
+  });
+  if (addAgreementError) console.log(addAgreementError);
```

Comment:
Let's move this to group context

Reply:
@avirankatz Since it's not blocking functionality or operability, can this refactor be part of a follow-up task?